workflow/sync/semaphore.go

```diff
@@ -165,7 +176,8 @@ func (s *PrioritySemaphore) tryAcquire(holderKey string) (bool, string) {
 	if s.pending.Len() > 0 {
 		item := s.pending.peek()
 		nextKey = fmt.Sprintf("%v", item.key)
-		if holderKey != nextKey {
+		if holderKey != nextKey && !isSameWorkflowNodeKeys(holderKey, nextKey) {
```

Comment:
We don't want to prioritize/order the same workflow nodes. The executetemplate will make sure the template order.

if the workflow has multiple nodes to wait for lock, the operator node order and semaphore queue order are going out of sync. if it same workflow key